cli/module_registry.go

```diff
@@ -380,6 +399,67 @@ func getNextModuleUploadRequest(file *os.File) (*apppb.UploadModuleFileRequest,
 	}, nil
 }
 
+func validateModuleFile(client *viamClient, moduleID moduleID, tarballPath, version string) error {
+	getModuleResp, err := client.getModule(moduleID)
+	if err != nil {
+		return err
+	}
+	entrypoint, err := getEntrypointForVersion(getModuleResp.Module, version)
+	if err != nil {
+		return err
+	}
+	//nolint:gosec
+	file, err := os.Open(tarballPath)
+	if err != nil {
+		return err
+	}
+	// TODO(APP-2226): support .tar.xz
+	if !strings.HasSuffix(file.Name(), ".tar.gz") {
```

Comment:
nit: could we do this by just checking the tarball path? instead of opening the file?

Reply:
We already have to open the file (because we are going to read the contents). Also, we want to do this because that way you can upload from symlinks to .tar.gz files and this will still work